lib/route-recognizer.js

```diff
@@ -164,12 +164,20 @@ function State(charSpec) {
   this.charSpec = charSpec;
   this.nextStates = [];
   this.charSpecs = {};
-  this.regex = undefined;
+  this.pattern = undefined;
+  this._regex = undefined;
   this.handlers = undefined;
   this.specificity = undefined;
 }
 
 State.prototype = {
+  regex: function () {
```

Comment:
It was, I didn't check but these maybe cacheable, the patterns maybe repeated, I didn't spend a lot of time investigating as the urls in the benchmark are a pretty lame sample set.